moto/sagemaker/models.py

```diff
@@ -3404,6 +3420,8 @@ def create_model_package(
                 if x.model_package_group_name == model_package_group_name
             ]
             model_package_version = len(model_packages_for_group) + 1
+            model_package_name = model_package_group_name
+        assert model_package_name
```

Comment:
There shouldn't be any assertions in the logic. If it's a valid scenario where the user hasn't provided the right values we can raise a proper exception, otherwise we should trust the logic above that the value is always set

Reply:
Hmm, this was just for mypy type narrowing, let me check if I can solve it other way.

Reply:
I've "solved" this with type casting for passing it to the `ModelPackage` class. This is because the `model_package_name` as the function parameter is `Optional`, and because of that even after we treat the case with the if-statement, it still is considered `str | None`.

So we cast it to an `str` after the if-statement.

What do you think?